comp/metadata/host/hostimpl/utils/meta.go

```diff
@@ -63,13 +65,19 @@ func GetMeta(ctx context.Context, conf model.Reader) *Meta {
 	}
 
 	m := &Meta{
-		SocketHostname: osHostname,
-		Timezones:      []string{tzname},
-		SocketFqdn:     util.Fqdn(osHostname),
-		EC2Hostname:    ec2Hostname,
-		HostAliases:    cloudproviders.GetHostAliases(ctx),
-		InstanceID:     instanceID,
-		AgentHostname:  agentHostname,
+		SocketHostname:            osHostname,
+		Timezones:                 []string{tzname},
+		SocketFqdn:                util.Fqdn(osHostname),
+		EC2Hostname:               ec2Hostname,
+		HostAliases:               cloudproviders.GetHostAliases(ctx),
+		InstanceID:                instanceID,
+		AgentHostname:             agentHostname,
+		HostnameResolutionVersion: "1.0",
+	}
+
+	legacyResolutionHostnameData, _ := hostname.GetWithProviderWithLegacyHostnameResolution(ctx)
```

Comment:
I agree, I applied this change,
other proposition: `GetWithProviders` becomes a parent function that calls either the legacy resolution or the other one, it would make some refactoring so if we want to do this maybe keep it for an other PR ?